Adds dns-prefetch to SMF.

WhistlingDonkey opened the pull request:
Adds a dns-prefetch list and function to output a list of dns-prefetch
link-rels in the document head. dns-prefetch can assist with page
loading times by attempting to pre resolve domain names before the
resource at the domain is requested in the DOM build.

Signed by WhistlingDonkey, github@pyro.life

WhistlingDonkey committed:
Adds dns-prefetch to SMF.

Adds a dns-prefetch list and function to output a list of dns-prefetch
link-rels in the document head. dns-prefetch can assist with page
loading times by attempting to pre resolve domain names before the
resource at the domain is requested in the DOM build.

Signed by WhistlingDonkey, github@pyro.life

Sesquipedalian commented:
Please open an issue explaining why these changes are required (or at least recommended).

WhistlingDonkey committed:
Fixes missing check on $modSettings['dns_prefetch_domains']

Signed by WhistlingDonkey, github@pyro.life

WhistlingDonkey committed:
- Changed input type to large_text
- Changed domain list to newline seperated instead of comma seperated as it's
easier to work with on screen
- rewrote parsing function to use newline instead of commas
- Changed Help text
- Added preconnect

Signed WhistlingDonkey github@pyro.life

WhistlingDonkey commented:
I think this is the only way I can link the issue to this...
#6912

WhistlingDonkey commented:
Closing this Pull request. I'm not happy with it and have come up with a better way.
